### PR TITLE
ENH: add option to show by coordinate in DiscreteLp

### LINKED ISSUE #260: Alternative to indices in show
We strive quite hard to allow users to think continuous, the `indices=...` concept kindoff breaks this, and is often hard to think about (at least I have to do some mental gymnastics each time I call it for nontrivial cases).

How about adding some "continuous" syntax, that translates to indices. Some examples of how this could look is:

```python
# Make space
X = uniform_discr([-1, -1], [1, 1], 157)
x = X.element(...)

# Current
x.show(indices=np.s_[:, 78])  # slice through the middle

# Some alternatives
x.show(values=[None, 0.0])
x.show(values=np.s_[:, 0.0])
x.show(y=0.0)
```

### PR DESCRIPTION
**Pull request by adler-j**
This closes #260

### CODE REVIEW

File: examples/visualization/show_2d.py [hunk 1]
```diff
@@ -28,5 +28,11 @@
...
 
-# Can also force "instant" plotting
+# Show all data
 vec.show(show=True)
+
+# We can show subsets by index:
+vec.show(indices=[slice(None), 50])
+
+# Or we can show by coordinate
+vec.show(coords=[None, 0.5])
```
Comment (adler-j) [inline examples/visualization/show_2d.py]: I prefer this syntax so much to the indices guy, we should perhaps try to modify all examples where it occurs aswell. This is much more "odly".

File: odl/discr/lp_discr.py [hunk 2]
```diff
@@ -688,6 +697,26 @@ def show(self, title=None, method='', indices=None, show=False, fig=None,
...
+            interv = self.space.uspace.domain
+            shape = self.shape
+            indices = []
+            for begin, end, n, coord in zip(interv.begin, interv.end,
+                                            shape, coords):
+                if coord is None:
+                    indices += [slice(None)]
+                else:
+                    if not begin <= coord <= end:
+                        raise ValueError('Coord {} not in range {}'
```
Comment (kohr-h) [inline odl/discr/lp_discr.py]: "coord"

File: odl/discr/lp_discr.py [hunk 3]
```diff
@@ -688,6 +697,26 @@ def show(self, title=None, method='', indices=None, show=False, fig=None,
...
+                                         ''.format(coord, [begin, end]))
+
+                    # Todo: handle end = begin
```
Comment (kohr-h) [inline odl/discr/lp_discr.py]: Throw an error? I guess using `None` is the only way not to run into rounding errors.

**Comment by adler-j** [comment]
Making a new for merge into master

**Comment by adler-j** [comment]
Obviously closed the wrong one...

**Comment by adler-j** [comment]
Fixed comments, merge?

File: examples/visualization/show_2d.py (hunk 1 above)
Comment (kohr-h) [inline examples/visualization/show_2d.py]: odlarious maybe?

**Comment by kohr-h** [comment]
Looking good to me

**Pull request merged**